yarn-project/foundation/src/aztec-address/index.ts

```diff
@@ -69,8 +69,17 @@ export class AztecAddress {
     return new AztecAddress(hexToBuffer(buf));
   }
 
+  /**
```

Comment:
Also (I can't comment on unchanged code), shouldn't the constructor of `Address` be ensuring the `value` passed in is a valid x-coordinate? And perhaps the name of the property `value` should be `x` (or something), to hint at the fact this value gets converted into a point.
Also also, shouldn't this `Address` class have a `toAddressPoint` method?
The doc comment at the top of the `Address` class should also provide more details relating to points.

Reply:
Also also, there's a `zero()` method, but is `0` a valid x-coordinate? Does -17 have a square root in the field?

Reply:
> And it's also fine to use invalid addresses
Is it?

I just worry that if the constructor doesn't enforce that the address is valid, then a user might create an invalid address and send funds to it. (Sure, the contract facilitating the transfer should have constrains to prevent this, but defense in depth, perhaps?).
Maybe Aztec's "zero address" should be `5` (or whatever the smallest valid x is)?